### DISCUSSION TITLE
"Artifact update failure" on github pipelines

### DISCUSSION THREAD

**Discussion by diboetkerdigital** [renovatebot/renovate#9846] [Request Help]
**What Renovate type, platform and version are you using?**
GitHub app.

**Describe the bug**
Renovate opens a PR, composer.json has suggested changes
The posted comment in the PR shows this description:

> ###  Artifact update problem
> Renovate failed to update an artifact related to this branch. You probably do not want to merge this PR as-is.
> 
>  Renovate will retry this branch, including artifacts, only when one of the following happens:
> 
> * any of the package files in this branch needs updating, or
> * the branch becomes conflicted, or
> * you check the rebase/retry checkbox if found above, or
> * you rename this PR's title to start with "rebase!" to trigger it manually
> 
> The artifact failure details are included below:
> 
> ##### File name: pimcore/composer.lock
> ```
> Command failed: docker run --rm --name=renovate_composer --label=renovate_child -v "/mnt/renovate/gh/[project_path]":"/mnt/renovate/gh/[project_path]" -v "/tmp/renovate-cache":"/tmp/renovate-cache" -e COMPOSER_CACHE_DIR -e COMPOSER_AUTH -w "/mnt/renovate/gh/[project_path]/pimcore" docker.io/renovate/composer:1.10.21 bash -l -c "composer update cache/integration-tests dama/doctrine-test-bundle doctrine/doctrine-fixtures-bundle nelmio/api-doc-bundle pimcore/pimcore snc/redis-bundle wikimedia/composer-merge-plugin --with-dependencies --ignore-platform-reqs --no-ansi --no-interaction --no-scripts --no-autoloader"
> Loading composer repositories with package information
> Updating dependencies (including require-dev)
> PHP Fatal error:  Allowed memory size of 1610612736 bytes exhausted (tried to allocate 4096 bytes) in phar:///usr/local/composer/1.10.21/bin/composer/src/Composer/DependencyResolver/Rule2Literals.php on line 53
> ```

**Additional context**
It happens to occur in every PR that is related to updating dependencies.

i seems like it's running out of memory, not sure what i can do about that.

**Comment by rarkins** [comment]
I haven't seen that before, but it definitely looks like it's out of memory. We don't have the ability to increase memory further on the workers we use so you may need to consider self-hosted if there's no way to reduce composer memory use through config.

**Reply by diboetkerdigital** [reply]
Yeah at the moment the only workaround i thought about was to do it manually. composer's memory problem was noticeable and I also had to increase the composer memory limit in the container.
I guess that could make an interesting use case to consider in the future

**Reply by rarkins** [reply]
Our machines have 4GB RAM and in theory there should not be too much else going on at the same time Composer is running, so maybe we have a way of reconfiguring either Docker or Composer to allocate more memory.

**Reply by mikepsinn** [reply]
@rarkins That would be awesome if you could increase it!

Are you able to prefix some PHP arguments when you call your `composer update` command like:
`php -d memory_limit=-1 composer.phar ...` 
or 
`php composer.phar COMPOSER_MEMORY_LIMIT=-1 ...`
?

https://stackoverflow.com/questions/49212475/composer-require-runs-out-of-memory-php-fatal-error-allowed-memory-size-of-161

Thanks for all your great work! :D

**Reply by viceice** [reply]
The env vars looks good, should be easy to add. 🤗
